backup/barman-montagu

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python3
+"""Set up and use barman (Postgres streaming backup) for montagu
+
+Usage:
+  barman-montagu setup [options] <host>
+  barman-montagu status
+  barman-montagu barman [--] [<args>...]
+  barman-montagu recover [--wipe-first] [<backup-id>]
+  barman-montagu wipe-recover
+  barman-montagu destroy
+
+Options:
+  --configure-cron          Configure a once-a-minute cron job
+  --password-group=<group>  Password group [default: production]
+  --image-tag=<tag>         Barman image tag [default: master]
+  --pull-image              Pull image before running?
+  --no-clean-on-error       Don't clean up container/volumes on error
+"""
+
+import os
+import re
+import socket
+import subprocess
+
+import crontab
+import docker
+import docopt
+import hvac
+
+# These defaults are all fairly montagu specific
+DEFAULT_VAULT_ADDR = "https://support.montagu.dide.ic.ac.uk:8200"
+BARMAN_CONTAINER = 'barman-montagu'
+BARMAN_VOLUMES = {'data': 'barman_data',
+                  'logs': 'barman_logs',
+                  'recover': 'barman_recover'}
+DATABASE_NAME = "montagu"
+DOCKER_REGISTRY = "docker.montagu.dide.ic.ac.uk:5000"
+DOCKER_IMAGE = "montagu-barman"
+PATH_CRON = "/etc/cron.d/barman-montagu"
+
+
+def setup(host, password_group, tag, pull, cron, clean_on_error):
+    start_barman(host, password_group, tag, pull, clean_on_error)
+    if cron:
+        configure_cron()
+
+
+def status():
+    barman = get_barman_container(error=False)
+    if barman:
+        cron_status = "enabled" if os.path.exists(PATH_CRON) else "disabled"
+        print("{} is running".format(BARMAN_CONTAINER))
+        print("cron job {}".format(cron_status))
+        exec_safely(barman, ["barman", "status", DATABASE_NAME],
+                    check=True, stream=True)
+    else:
+        print("{} is not running".format(BARMAN_CONTAINER))
+
+
+def barman(args):
+    barman = get_barman_container()
+    exec_safely(barman, ["barman"] + args, check=True, stream=True)
+
+
+def recover(id, wipe_first):
+    barman = get_barman_container()
+    if not id:
+        id = "latest"
+    if wipe_first:
+        wipe_recover(barman)
+    exec_safely(barman, ["barman", "recover", DATABASE_NAME, id, "/recover"],
```

Comment:
Should this use `BARMAN_VOLUMES` rather than a bare path to `/recover`?

Reply:
No I see, BARMAN_VOLUMES is the volume names for the host system. They get mapped to internal paths (which is what this is) in `start_barman`. Maybe add a comment to the `BARMAN_VOLUMES` declaration to make its role clear?

Reply:
This is confusing and I've reworked it